code/game/objects/structures/flora.dm

```diff
@@ -78,6 +78,15 @@
 	icon = 'icons/obj/plants.dmi'
 	icon_state = "plant-26"
 
+//Added random icon selection for potted plants.
+//It was silly they always used the same sprite when we have 26 sprites of them in the icon file
+/obj/structure/flora/pottedplant/New()
+	..()
+	var/number = rand(1,26)

```

Comment:
Backing up Fowl here. Random icon generation like that is a bitch when mapping.

If you absoloutely need this, make a new child with type `/obj/structure/flora/pottedplant/random` and go nuts with that.